servicecombkie/kie_client.go

```diff
@@ -0,0 +1,197 @@
+		return nil, err
+	}
+	for _, doc := range configurationsValue {
+		for _, kvDoc := range doc.Data {
+			if key == kvDoc.Key {
+				openlogging.GetLogger().Debugf("The Key Value of : ", kvDoc.Value)
+				return doc, nil
+			}
+		}
+	}
+	return nil, errors.New("can not find value")
+}
+
+//PullConfigsByDI not implemented
+func (kieClient *Client) PullConfigsByDI(dimensionInfo string) (map[string]map[string]interface{}, error) {
+	// TODO Return the configurations for customized Projects in Kie Configs
+	fmt.Println(" PullConfigsByDI")
+	return nil, errors.New("not implemented")
+}
+
+//PushConfigs put config in kie by key and labels.
+func (kieClient *Client) PushConfigs(data map[string]interface{}, serviceName, version, app, env string) (map[string]interface{}, error) {
+	fmt.Println("PushConfigs")
+	var configReq model.KVDoc
+	labels := map[string]string{"servicename": serviceName, "version": version, "app": app, "env": env}
+	configResult := make(map[string]interface{})
+	for key, configValue := range data {
+		configReq.Key = key
+		configReq.Value = configValue.(string)
+		configReq.Labels = labels
+		configurationsValue, err := kieClient.KieClient.Put(context.TODO(), configReq, sckieclient.WithProject(serviceName))
+		if err != nil {
+			openlogging.GetLogger().Error("Error in PushConfigs to Kie: " + err.Error())
+			return nil, err
+		}
+		openlogging.GetLogger().Debugf("The Key Value of : ", configurationsValue)
+		configResult[configurationsValue.Key] = configurationsValue.Value
+	}
+	return configResult, nil
+}
+
+//DeleteConfigsByKeys use keyId for delete
+func (kieClient *Client) DeleteConfigsByKeys(keys []string, serviceName, version, app, env string) (map[string]interface{}, error) {
+	fmt.Println("DeleteConfigsByKeys")
+	result := make(map[string]interface{})
+	for _, keyId := range keys {
+		err := kieClient.KieClient.Delete(context.TODO(), keyId, "", sckieclient.WithProject(serviceName))
+		if err != nil {
+			openlogging.GetLogger().Errorf("Error in Delete from Kie. %s " + err.Error())
+			result[keyId] = RetDelErr
+		} else {
+			openlogging.GetLogger().Debugf("Delete The KeyId:%s", keyId)
+			result[keyId] = RetSucc
+		}
+	}
+	return result, nil
+}
+
+//Watch not implemented because kie not support.
+func (kieClient *Client) Watch(f func(map[string]interface{}), errHandler func(err error)) error {
+	return nil
+}
+
+//Options.
+func (kieClient *Client) Options() config.Options {
+	fmt.Println("Options")
+	optionInfo := config.Options{
+		ServiceName:   kieClient.serviceName,
+		Version:       kieClient.version,
+		Endpoint:      kieClient.URI,
+		EnableSSL:     kieClient.EnableSSL,
+		AutoDiscovery: kieClient.AutoDiscovery,
+		Namespace:     kieClient.Namespace,
+		TenantName:    kieClient.TenantName,
+	}
+	return optionInfo
+}
+
+//InitConfigKie initialize the Kie Client
+func InitConfigKie(options config.Options) (config.Client, error) {
```

Comment:
merge InitConfigKie  and NewKieClient into one method "NewClient"

Reply:
Edit